Task iterated

SAMask127node opened the pull request.

SAMask127node committed:
Task iterated

sergeypil commented on src/main/java/com/epam/gym/service/TraineeService.java:
```diff
@@ -0,0 +1,72 @@
+import java.util.Optional;
+
+@Service
+public class TraineeService {
```
Nit: Usually it is good practice to introduce interfaces wth contracts and then extends them

SAMask127node replied:
interfaces introduced and implemented
please see new [commit](https://github.com/SAMask127node/gym-crm-spring-core-project1/pull/1/commits/a22fbb859409f74d8ae7f845468c8804455d82e1)

sergeypil commented on src/main/java/com/epam/gym/service/TraineeService.java:
```diff
@@ -0,0 +1,72 @@
+public class TraineeService {
+    private static final Logger logger = LoggerFactory.getLogger(TraineeService.class);
+
+    private final Map<String, Trainee> traineeMap;
```
I think we need to add DAO layer

SAMask127node replied:
Dao layer was added
please see new [commit](https://github.com/SAMask127node/gym-crm-spring-core-project1/pull/1/commits/a22fbb859409f74d8ae7f845468c8804455d82e1)

sergeypil commented on src/test/java/com/epam/gym/service/TrainerServiceTest.java:
```diff
@@ -0,0 +1,60 @@
+        Trainer t = trainerService.create("Grace", "Hopper");
+        assertNotNull(t, "Returned Trainer should not be null");
+        assertEquals("Grace.Hopper", t.username(), "Username should be first.last");
+        assertNotNull(t.password(), "Password should be non‐null");
```
I'd recommend to use asserj lib for assertions

SAMask127node replied:
asserj lib for assertions was used
please see new [commit](https://github.com/SAMask127node/gym-crm-spring-core-project1/pull/1/commits/a22fbb859409f74d8ae7f845468c8804455d82e1)

sergeypil commented on src/test/java/com/epam/gym/service/TrainerServiceTest.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Test
+    void testCreateAndFind() {
+        Trainer t = trainerService.create("Grace", "Hopper");
```
I'd recommend to add //given //when //then comments to tests

SAMask127node replied:
//given //when //then comments were added to tests
please see new [commit](https://github.com/SAMask127node/gym-crm-spring-core-project1/pull/1/commits/a22fbb859409f74d8ae7f845468c8804455d82e1)

sergeypil commented on src/test/java/com/epam/gym/service/TrainingServiceTest.java:
```diff
@@ -0,0 +1,47 @@
+/**
+ * Verifies TrainingService functionality using an in‐memory map (StorageConfig).
+ */
+@SpringJUnitConfig(StorageConfig.class)
```
This is integration test.
I'd recommend to use @ExtendWith(MockitoExtension.class) and unit test. In the tasks we need unit tests.

SAMask127node replied:
Only unit tests were left

SAMask127node committed:
refactor: best principles used

rewritten with Interfaces, Dao layer added, //given //when //then comments added to tests, asserj lib for assertions used,

SAMask127node committed:
refactor: best principles used

rewritten with Interfaces, Dao layer added, //given //when //then comments added to tests, asserj lib for assertions used,

SAMask127node committed:
first iteration

SAMask127node committed:
Merge branch 'develop' into feature/JPA

SAMask127node committed:
Merge pull request #2 from SAMask127node/feature/JPA

Feature/jpa
